### ISSUE TITLE
terraform-providers adoption tracking issue

### ISSUE THREAD

**Issue by VoyTechnology** [terra-farm/terraform-provider-virtualbox#50]
The purpose of this issue is to track the progress of possible adoption of this repository by the official [terraform-providers][1] repository. This should be the place to discuss what else is needed before this repository is needed. Additonally a milestone and a github project to track the progress after or while the exact requirements are discussed.

[1]: https://github.com/terraform-providers

**Comment by VoyTechnology** [comment]
Here are a couple of my suggestions:

1. Establish the owners and official maintainers of the repository, and document it in README for transparency.
2. Improve log transparency. #44 adds logs propagation which should make it easier to see the cause of an issue directly when running `terraform plan|apply`.
3. Add image checksum validation
4. Allow different methods of fetching the images. After #40 the supported types would be HTTP/HTTPS and locally. We should at the very least also add the most common cloud ones like AWS and GCP.
5. Completely deprecate `dep` and just use go modules. (#38)
6. __Add tests__

This is all I can think for now, but please let me know what do you think. Because of point 1 I don't even know who to tag :)

**Comment by ringods** [comment]
@VoyTechnology why the need to want this migrated? We haven't even been able to make the `terra-farm` organisation a place for providers to see the light and have shared maintainership. I don't know if moving this to Hashicorp would bring us anything additional.

**Comment by VoyTechnology** [comment]
Misread https://github.com/terra-farm/terraform-provider-virtualbox/issues/48#issuecomment-456199009. Replaced `terra-farm` with `terra-providers` in my head.

But I think the TODO list still holds overall for v1.0 release.

**Comment by VoyTechnology** [comment]
Feel free to close this issue if you think this is unnecessary

**Comment by ringods** [comment]
@VoyTechnology true regarding the todo list. Can you update this issue by:
* putting the todo list in the initial comment of this issue
* renaming the issue to "Prepare for v1.0 with TF 0.12" or somethign like that.

**Comment by VoyTechnology** [comment]
Updated the comment. Please let me know which of the points from https://github.com/terra-farm/terraform-provider-virtualbox/issues/50#issuecomment-456506855 I should add to the TODO list.

**Comment by ringods** [comment]
@VoyTechnology I updated the Terraform build dependency to v0.12.0. Needs some manual testing before we release a first provider version for TF 0.12.

**Comment by ringods** [comment]
@VoyTechnology I suggest we decouple the versioning towards v1 from the compatibility with TF 0.12. I can release a TF 0.12 compatible version shortly, but the provider lacks too much regarding datasources and/or resources that I would already like to make it a v1. 

OK if I create a release for TF 0.12 where users can file bugs against? From there, we can augment the provider to make a V1 ready.

Comments?

**Comment by VoyTechnology** [comment]
Makes sense, since TF 0.12 is already out. I think we should have some release ASAP.

**Comment by ringods** [comment]
@VoyTechnology the TF 0.12 compatible version is here:

https://github.com/terra-farm/terraform-provider-virtualbox/releases/tag/v0.2.0

**Comment by VoyTechnology** [comment]
Because the v1.0 is a bit ambiguous, for the time being I have split it into 2 releases which will come before this:

- [v0.3 - Storage](https://github.com/terra-farm/terraform-provider-virtualbox/milestone/2)
- [v0.4 - Networking](https://github.com/terra-farm/terraform-provider-virtualbox/milestone/3)

This split allows to prioritise. This doesn't mean that the networking is less important than storage, but it just gives some notion of a goal.
